### PR TITLE
🎨  public client registration updates

### PR DESCRIPTION
**Pull request by kirrg001**
refs #7654 

We have to adapt some changes to the public client registration.
Especially: updating data like blog title or urls.

In this PR i will push each change (each issue of the referred issues) into a separate commit.

### CODE REVIEW

**Comment by kirrg001** [comment]
Pushed my updates. I still need to final test my changes and add 1-2 tests.
Will do that in the tomorrow as first thing.

File: core/server/data/schema/schema.js [hunk 1]
```diff
@@ -172,6 +172,7 @@ module.exports = {
         slug: {type: 'string', maxlength: 150, nullable: false, unique: true},
         secret: {type: 'string', maxlength: 150, nullable: false},
         redirection_uri: {type: 'string', maxlength: 2000, nullable: true},
+        blog_uri: {type: 'string', maxlength: 2000, nullable: true},
```
Comment (ErisDS) [inline core/server/data/schema/schema.js]: I don't see the need for sorting this value locally? It should be the same as config.url.

Comment (kirrg001) [inline core/server/data/schema/schema.js]: Hmm I've added the `blog_uri` to the schema to be able to track a change.
So if config.url changes, how do we know?Maybe i am blind to see there is an easier solution 😬

**Comment by kirrg001** [comment]
Ready for review. 
If we have time i would like to get a merge of #7692. If not, no problem we can do that later.

**Comment by kirrg001** [comment]
I have to push one little update, but this is not related to the changed logic. The logic itself can be reviewed.

It's that passport-ghost is now responsible for doing retries. And the retry logic was removed in Ghost. I want to handle the auth.init as asynchronous operation. Pushing a separate commit soon.

**Comment by kirrg001** [comment]
I've added a new commit, see https://github.com/TryGhost/Ghost/pull/7690/commits/4a6ffa3af866f748e294c2e04581edc62a1aaded

We could think about outsourcing this commit to a separate PR.
So when you start Ghost: auth.init won't block your server.
It runs in background. If it can't register a client, passport-ghost will retry under the hood and if an error occurs you will see it in your log. You can always enable DEBUG=passport-ghost:* to see what is going on.

File: core/server/data/schema/schema.js (hunk 1 above)
Comment (ErisDS) [inline core/server/data/schema/schema.js]: Ok, I see where you are coming from. However, this field still doesn't feel right to me.

All of the other clients e.g. ghost-frontend are things that allow 3rd parties to talk to Ghost. 

This is the first client we have introduced that lets Ghost talk to a 3rd party. 

We will have more things like this, but none of them will have a "blog_uri", they may have another field.

If you look at reference material for similar systems: e.g. twitter's create an application, or GitHub, or Google API console, can you perhaps find a way to make this field work for more general cases than just this one?

**Comment by ErisDS** [comment]
Just one comment in follow up to the existing conversation - everything else is looking great 👍

File: core/server/data/schema/schema.js (hunk 1 above)
Comment (kirrg001) [inline core/server/data/schema/schema.js]: Ok understand. Good idea.
I looked at some references.

https://developers.google.com/identity/protocols/OAuth2UserAgent
https://developer.github.com/v3/oauth/
https://developers.facebook.com/docs/facebook-login/manually-build-a-login-flow

Nobody uses or defines more then the default oauth2 spec parameters (redirect_uri, scope, client_id, client_secret, state). 

Then i looked into the RFC spec again
https://tools.ietf.org/html/rfc7591

This is an example request to register a client (https://tools.ietf.org/html/rfc7591#page-16)
```
     POST /register HTTP/1.1
     Content-Type: application/json
     Accept: application/json
     Host: server.example.com

     {
      "redirect_uris": [
        "https://client.example.org/callback",
        "https://client.example.org/callback2"],
      "client_name": "My Example Client",
      "client_name#ja-Jpan-JP":
         "\u30AF\u30E9\u30A4\u30A2\u30F3\u30C8\u540D",
      "token_endpoint_auth_method": "client_secret_basic",
      "logo_uri": "https://client.example.org/logo.png",
      "jwks_uri": "https://client.example.org/my_public_keys.jwks",
      "example_extension_parameter": "example_value"
     }
```
^ BTW: they suggest using `client_name` and not `name`.

And they have listed more attributes here https://tools.ietf.org/html/rfc7591#page-9

```
   client_uri
      URL string of a web page providing information about the client.
      If present, the server SHOULD display this URL to the end-user in
      a clickable fashion.  It is RECOMMENDED that clients always send
      this field.  The value of this field MUST point to a valid web
      page.  The value of this field MAY be internationalized, as
      described in Section 2.2.
```

So i would change it to `client_uri` if you agree 👍

Comment (ErisDS) [inline core/server/data/schema/schema.js]: 👍

**Comment by kirrg001** [comment]
I will add one more small commit to fix a TODO, then we can merge.
see https://github.com/TryGhost/Ghost/pull/7690/files#diff-79a58308a1a36ab1b54544a17f88997aR125

**Pull request merged**